include/fastdds/dds/domain/DomainParticipant.hpp

```diff
@@ -367,6 +367,10 @@ class RTPS_DllAPI DomainParticipant
 
     fastrtps::rtps::ResourceEvent& get_resource_event() const;
 
+    const fastrtps::rtps::RTPSParticipant* rtps_participant() const;
```

Comment:
As commented with you, modified to provide to the RTPSParticipant a callback that will be used by the PDP in case it needs to check if a type is registered.